hedgehog/src/Hedgehog/Internal/State.hs

```diff
@@ -776,7 +773,7 @@ data ActionCheck state =
     }
```

Comment:
```
-- | Re-validations needed for a given `Action` after a shrink.
```

Reply:
I also wonder if `ctx1 { contextState = contextState ctx0 }` is the best way of describing what happens.
I understand that we generate commands with gen `commands`  from state in `ctx0`, in both branches.
Why would we use `ctx1` to start the second branch? Is it to avoid using the same variable names?|

If so, it is a hidden interaction that merits a justification in a comment.

Reply:
That sounds right, yeah. I haven't looked at that code in any depth. I don't want to add a bunch of comments to code that I haven't taken the time to fully understand, especially not in a PR that doesn't touch that code.